lib/staging/tls/tls.hpp

```diff
@@ -221,6 +222,19 @@ class Connection {
      * \note the certificate must not be freed
      */
     [[nodiscard]] const Certificate* peer_certificate() const;
+
+    /**
+     * \brief obtain the underlying SSL context
+     * \returns the underlying SSL context pointer
+     */
+    [[nodiscard]] SSL* ssl_context() const;
```

Comment:
The idea was to avoid making OpenSSL internals available.
shouldn't be needed since both read() and write() return
        want_read,  //!< non-blocking - operation waiting for read available on socket
        want_write, //!< non-blocking - operation waiting for write available on socket
(see above)

Reply:
Agreed that this should not be exposed. Given the temporary nature of IsoMux module (until the -2/DIN spec implementation in libiso15118 is ready) I marked this (and the set_timeout) function as deprecated so that they are not used by any other module. They will be removed together with IsoMux module then.